packages/cojson/src/coValues/coMap.ts

```diff
@@ -232,36 +188,64 @@ export class RawCoMapView<
    *
    * @category 1. Reading */
   keys<K extends keyof Shape & string = keyof Shape & string>(): K[] {
-    return (Object.keys(this.latest) as K[]).filter((key) => {
-      const latestChange = this.latest[key];
+    return (Object.keys(this.ops) as K[]).filter((key) => {
+      const entry = this.getRaw(key);
 
-      if (!latestChange) {
+      if (entry === undefined) {
         return false;
       }
 
-      if (latestChange.change.op === "del") {
+      if (entry.change.op === "del") {
         return false;
-      } else {
-        return true;
       }
+
+      return true;
     });
   }
 
+  getRaw<K extends keyof Shape & string>(key: K) {
+    let latestChange = this.latest[key];
+
+    if (latestChange === undefined) {
+      const entries = this.ops[key];
+
+      // Time travel values are lazily computed
+      if (entries && !(key in this.latest)) {
```

Comment:
This was actually my main goal.

With the `writeOnly` role and the Inbox pattern we will probably have groups with a huge number of keys, so I wanted to make time travel in that setting as fast as possible